addons/binding/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,440 @@
+                            .filter(channelUID -> isLinked(channelUID)
+                                    && CHANNEL_GROUP_DAILY_FORECAST_DAY3.equals(channelUID.getGroupId()))
+                            .forEach(channelUID -> {
+                                State state = getValue(channelUID.getIdWithoutGroup(), hashParameters,
+                                        timeSeries.get(0).getValidTime());
+                                updateState(channelUID, state);
+                            });
+                }
+
+                temp = hashParameters.get(CHANNEL_TEMPERATURE_JSON).getValues()[0];
+
+                if (cdf.format(currentvaliddate).equals(cdf.format(timeSeries.get(0).getValidTime()))) {
+                    if (temp < tmin) {
+                        tmin = temp;
+                    }
+                    if (temp > tmax) {
+                        tmax = temp;
+                    }
+                } else if (cdf.format(tomorrowvaliddate).equals(cdf.format(timeSeries.get(0).getValidTime()))) {
+                    if (temp < tmin1) {
+                        tmin1 = temp;
+                    }
+                    if (temp > tmax1) {
+                        tmax1 = temp;
+                    }
+                } else if (cdf.format(day2validdate).equals(cdf.format(timeSeries.get(0).getValidTime()))) {
+                    if (temp < tmin2) {
+                        tmin2 = temp;
+                    }
+                    if (temp > tmax2) {
+                        tmax2 = temp;
+                    }
+                } else if (cdf.format(day3validdate).equals(cdf.format(timeSeries.get(0).getValidTime()))) {
+                    if (temp < tmin3) {
+                        tmin3 = temp;
+                    }
+                    if (temp > tmax3) {
+                        tmax3 = temp;
+                    }
+                }
+
+                parameters.clear();
+                hashParameters.clear();
+                timeSeries.remove(0);
+            }
+
+            // update channels min max
+            logger.debug("Updating min/max channels");
+            updateState("current#temperature-min", new DecimalType(tmin));
+            updateState("current#temperature-max", new DecimalType(tmax));
+            updateState("forecastTomorrow#temperature-min", new DecimalType(tmin1));
+            updateState("forecastTomorrow#temperature-max", new DecimalType(tmax1));
+            updateState("forecastDay2#temperature-min", new DecimalType(tmin2));
+            updateState("forecastDay2#temperature-max", new DecimalType(tmax2));
+            updateState("forecastDay3#temperature-min", new DecimalType(tmin3));
+            updateState("forecastDay3#temperature-max", new DecimalType(tmax3));
+        } catch (final Exception e) {
+            logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
How is the thing coming back online again?

Reply:
Not a clue :)

Reply:
The question here is: Will the scheduler finish or continue? If it continues, just set the status to ONLINE after a successfull refresh. If the scheduler exits, you should schedule the task again (and then set the status to ONLINE after the first successfull refresh).

Reply:
What i want is that if something goes wrong it should just schedule the task again and try again after refresh time. But i don not know how to do this.